lib/Provider/Command/MessageSend.php

```diff
@@ -0,0 +1,155 @@
+		} catch (ClientException $e) {
+			throw new SendException('Error: occurred while retrieving mail account details', 0, $e);
+		}
+		// convert mail provider message to mail app message
+		$localMessage = new LocalMessage();
+		$localMessage->setType($localMessage::TYPE_OUTGOING);
+		$localMessage->setAccountId($account->getId());
+		$localMessage->setSubject((string)$message->getSubject());
+		$localMessage->setBody((string)$message->getBody());
+		// disabled due to issues caused by opening these messages in gui
+		//$localMessage->setEditorBody($message->getBody());
+		$localMessage->setHtml(true);
+		$localMessage->setSendAt($this->time->getTime());
+		// convert mail provider addresses to recipient addresses
+		$to = $this->convertAddressArray($message->getTo());
+		$cc = $this->convertAddressArray($message->getCc());
+		$bcc = $this->convertAddressArray($message->getBcc());
+		// save attachments
+		$attachments = [];
+		try {
+			foreach ($message->getAttachments() as $entry) {
+				$attachments[] = $this->attachmentService->addFileFromString(
+					$userId,
+					(string)$entry->getName(),
+					(string)$entry->getType(),
+					(string)$entry->getContents()
+				)->jsonSerialize();
+			}
+		} catch (UploadException $e) {
+			$this->purgeSavedAttachments($attachments);
+			throw new SendException('Error: occurred while saving mail message attachment', 0, $e);
+		}
+		// save message
+		try {
+			$localMessage = $this->outboxService->saveMessage(
+				$account,
+				$localMessage,
+				$to,
+				$cc,
+				$bcc,
+				$attachments
+			);
+		} catch (\Throwable $e) {
+			$this->purgeSavedAttachments($attachments);
+			throw new SendException('Error: occurred while saving mail message', 0, $e);
+		}
+		// send message
+		try {
+			$localMessage = $this->outboxService->sendMessage($localMessage, $account);
+		} catch (\Throwable $e) {
+			throw new SendException('Error: occurred while sending mail message', 0, $e);
+		}
+
+		return $localMessage;
+	}
+
+	/**
+	 * Converts IAddress objects collection to plain array
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param array<int,IAddress> $addresses collection of IAddress objects
+	 *
+	 * @return array<int, array{email: string, label?: string}> collection of addresses and labels
+	 */
+	protected function convertAddressArray(array $addresses): array {
+		return array_map(static function (IAddress $address) {
+			return !empty($address->getLabel())
+				? ['email' => (string)$address->getAddress(), 'label' => (string)$address->getLabel()]
+				: ['email' => (string)$address->getAddress()];
+		}, $addresses);
+	}
+
+	/**
+	 * Removes attachments from data store
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param array $attachments collection of local attachment objects
+	 */
+	protected function purgeSavedAttachments(array $attachments): void {
+		foreach ($attachments as $attachment) {
+			$this->attachmentService->deleteAttachment($attachment->getUserId(), $attachment->getId());
```

Comment:
getUserId and getId only work when you pass an LocalAttachment object. You are calling jsonSerialize in Line 95, and therefore we have a plain array here. 

I prefer working with real objects rather than plain arrays and therefore recommend removing jsonSerialize in Line 95, add the phpdoc back for purgeSavedAttachmends, and use ` array_map(static fn(LocalAttachment $attachment) => $attachment->jsonSerialize(), $attachments)`  in Line 109.